File: 1.py
def twoSum(nums, target):
    a = dict()
    for i, x in enumerate(nums):
       a[x] = i

    for i, x in enumerate(nums):
        complement = target - x
        if complement in a:
            val = a[complement]
            if val and val !=i:
                return [a[complement],i]


print(twoSum([2,5,5,11], 10))

File: 1.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {

	values := make(map[int]int)

	for i, v := range nums {
		values[v] = i
	}

	i := 0

	for ; i < len(nums); i++ {
		complement := target - nums[i]
		val, ex := values[complement]
		if ex && val != i {
			return []int{val, i}
		}
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{3, 2, 4}, 6))
}
